Check the implications of increasing the sample size when estimating the mean of a
variable. How does it affect the confidence interval width?

In [32]:
import numpy as np
import pandas as pd

In [19]:
df = pd.read_csv(r'E:\Code\Maths assignment\tested.csv')


In [20]:
# grouping data by first class tickets
newdf = df[df['Pclass'] == 1][['Fare']]
# Since very few(1) have paid a fare above 300, it's best to remove that value as it could drastically affect the mean.
newdf = newdf[newdf['Fare'] < 300]

In [ ]:
# Sampling 
samp1 = newdf['Fare'].sample(n=10, random_state=42)
samp2 = newdf['Fare'].sample(n=40, random_state=42)
samp3 = newdf['Fare'].sample(n=80, random_state=42)

In [ ]:
# calculating sample mean
Smean1=samp1.mean()
Smean2=samp2.mean()
Smean3=samp3.mean()

In [ ]:
print(Smean1)
print(Smean2)
print(Smean3)


96.90834
91.882395
90.6479175
Population mean- 90.33643962264152


In [ ]:
sigma = np.std(newdf['Fare'])
Zval = 1.96

In [ ]:
# CI For n = 10
SE1 = sigma / np.sqrt(10)
CI1_lower = Smean1 - Zval * SE1
CI1_upper = Smean1 + Zval * SE1
print(f"Confidence Interval for n=10: ({CI1_lower}, {CI1_upper})")

Confidence Interval for n=10: (51.08818943695371, 142.72849056304628)


In [ ]:
# Ci for n = 40
SE2 = sigma / np.sqrt(40)
CI2_lower = Smean2 - Zval * SE2
CI2_upper = Smean2 + Zval * SE2
print(f"Confidence Interval for n=40: ({CI2_lower}, {CI2_upper})")

Confidence Interval for n=40: (68.97231971847685, 114.79247028152315)


In [38]:
# CI for n = 80
SE3 = sigma / np.sqrt(80)
CI3_lower = Smean3 - Zval * SE3
CI3_upper = Smean3 + Zval * SE3
print(f"Confidence Interval for n=80: ({CI3_lower}, {CI3_upper})")

Confidence Interval for n=80: (74.44804791094069, 106.84778708905932)


As the sample size increases, the confidence interval becomes narrower.

For n=10, the interval is (51.09, 142.73), showing a wide range.
For n=40, the interval narrows to (68.97, 114.79).
For n=80, the interval further tightens to (74.45, 106.85).

This indicates that larger samples provide more precise estimates of the population mean, reducing uncertainty in our estimation.